ansible/library/extract_log.py

```diff
@@ -108,19 +109,23 @@ def extract_number(s):
         return int(ns[0])
 
 
-def convert_date(s):
+def convert_date(fct, s):
     dt = None
     re_result = re.findall(r'^\S{3}\s{1,2}\d{1,2} \d{2}:\d{2}:\d{2}\.?\d*', s)
     # Workaround for pytest-ansible
     loc = locale.getlocale()
     locale.setlocale(locale.LC_ALL, (None, None))
 
     if len(re_result) > 0:
-        str_date = re_result[0]
+        str_date = '{:04d} '.format(fct.year) + re_result[0]
         try:
-            dt = datetime.datetime.strptime(str_date, '%b %d %X.%f')
+            dt = datetime.datetime.strptime(str_date, '%Y %b %d %X.%f')
         except ValueError:
-            dt = datetime.datetime.strptime(str_date, '%b %d %X')
+            dt = datetime.datetime.strptime(str_date, '%Y %b %d %X')
+        if (fct - dt).days > 183:
+            dt.replace(year = dt.year + 1)
+        if (dt - fct).days > 183:
+            dt.replace(year = dt.year - 1)
```

Comment:
Hi, 183 is the number of days in half year. It's just like we check whether the different of two 32-bit unsigned integer larger than 2^(32-1) to judge that some kinds of wrap around happen. Just a fair choice.
For the concerned logic, you're right. I thought os.path.getctime() is getting the file creation time, but it seems that this is not the case in Unix/Linux system. What is your suggestion? remove it or keep it there just in case?

Reply:
I suggest to 
* add comments to the number `183`.
* remove the code path that won't be hit to avoid confusion
* add comments to explain the logic.

Thanks!

Reply:
Hi, if fct is always greater than ct, I guess the logic can be changed to
````
if dt > fct:
     dt.replace(year = dt.year - 1)
````
No need to use the number 183, right?

Reply:
I am not sure if there is any race condition that could cause exceptions. IMO, keep the 183 could avoid the race condition.

Reply:
Hi, done, please review again. Thanks!